### DISCUSSION TITLE
Cross-platform access to system user name?

### DISCUSSION THREAD

**Discussion by repi** [nushell/nushell#10979] [Q&A] [answered]
On Linux and Mac one can use `$env.USER` but on Windows that does not exist but there is an equivalent `$env.USERNAME`. 

is there a cross-platform command to access the currently active user that uses these underlying env vars? would be nice maybe in the `sys` command where you also have the `hostname`

**Comment by amtoine** [comment, marked as answer]
maybe `whoami` could do the job here? :yum:

**Reply by repi** [reply]
ah that does work! though it gives `<hostname>\<username>` so needs a some splitting 

I did a small hacky solution for now also:

```nu
try { 
        $env.USER # unix
} catch { 
        $env.USERNAME # windows
}
```

**Reply by fdncred** [reply]
you may not be using the latest version. I don't get host\user on my Windows box. Just the user. I'm on the latest main, which is 0.86.1.

**Reply by repi** [reply]
ah yes was on 0.84.0, will try latest for it. thanks

**Reply by repi** [reply]
confirmed! `whoami` on 0.86.1 / latest unreleased git version this gives only the user name on Windows. latest released 0.86.0 version gives the incorrect `<hostname>\<username>` on Windows (but works fine on Linux).

great then have both a short term workaround for latest published nu and a real solution once 0.86.1 is released. thanks everyone for super quick turnaround here!

**Reply by amtoine** [reply]
instead of 
```nushell
try { 
        $env.USER # unix
} catch { 
        $env.USERNAME # windows
}
```
you can do
```nushell
$env.USER? | default $env.USERNAME
```
:wink:

**Reply by repi** [reply]
nice, much cleaner. I'm a bit of a nushell noob but loving it so far ❤️ 
split my time working on our Rust projects on Mac, Linux and Windows, so having a cross-platform modern structured shell and scripting language is super nice.

**Reply by fdncred** [reply]
Don't feel bad @repi, I've been working on nushell longer than @amtoine and just a couple days ago he suggested the same thing for one of my scripts. LOL. @amtoine is great at nushell scripting 💥!!!
